include/functions_config.inc.php

```diff
@@ -1065,29 +1094,20 @@ function serendipity_getSessionLanguage() {
 function serendipity_getPostAuthSessionLanguage() {
     global $serendipity;
 
-    if (! is_null($serendipity['detected_lang'])) {
-        return $serendipity['detected_lang'];
-    }
+    // cached result from getSessionLanguage()
+    if (!empty($serendipity['detected_lang']) && !defined('IN_serendipity_admin')) return $serendipity['detected_lang'];
 
+    // if that returned the fallback language, try user default language or fallback
     if ($_SESSION['serendipityAuthedUser']) {
         if ($serendipity['expose_s9y']) serendipity_header('X-Serendipity-InterfaceLangSource: Database');
-        $lang = $serendipity['lang'];
-    } else {
-        $lang = (isset($_SESSION['serendipityLanguage'])) ? $_SESSION['serendipityLanguage'] : $serendipity['lang'];
-    }
-
-    if (!isset($serendipity['languages'][$lang])) {
-        $lang = $serendipity['lang'];
+        $lang = $serendipity['lang']; //$serendipity['lang'] is overwritten by load_configuration(user) and points to the user default language
     }
 
+    // save the result into SESSION
     $_SESSION['serendipityLanguage'] = $lang;
 
     if ($serendipity['expose_s9y']) serendipity_header('X-Serendipity-InterfaceLang: ' . $lang);
 
-    if ($lang != $serendipity['lang']) {
```

Comment:
Are there maybe plugins or other instances that actually work in $serendipity['content_lang']? It feels a bit dangerous to remove this variable completely?

Reply:
I did a full-text search in the core, core-plugins and additional plugins and it returned empty.

Reply:
Sometimes I am afraid of places where dynamic array access could be used. Like:

```
$whatToCheck = 'content';
if (condition()) {
    $whatToCheck = 'interface';
}
echo $serendipity[$whatToCheck . '_lang'];
```

I did learn that this is bad code, but it is still left in a couple of places, and impossible to search for. So I'm leaning toward the safer variant to leave the variable in there, it shouldn't hurt but shouldn't break things. Other opinions maybe? Of course the wish to only leave relevant code is important as well.

Reply:
I removed it and it didn't break anything. Because I started with the problems caused by inconsistent language settings, I would be glad if something shows up which depends on that variable because I need to fix that, too. I have so far condensed the language variables into these:
$serendipity['autolang'] : the name is misleading, this is the ultimate fallback language which remains 'en' in all cases. 
$serendipity['lang']: the language to be shown for every output
$serendipity['default_lang']: the default language of the blog
$serendipity['detected_lang'] the language determined by user input or browser negotiation, if that fails, NULL.
$serendipity['GET']['user_language'] the language requested for the blog by user
$serendipity['GET'['lang_selected'] the language requested for an single article by user

I'd prefer to clean it up as much as possible. If there is a call to that variable, this would be a very clear error message and a simple replace with one of the above.

Reply:
OK, you convinced me. Let's remove it.